### PR TITLE
[FLINK-25537][JUnit5 Migration] Module: flink-core with,Package: types

### PR DESCRIPTION
**Pull request by GOODBOY008**
Changes:

- Migrate Module: flink-core with,Package: types to junit5

### CODE REVIEW

**Comment by GOODBOY008** [comment]
@Jiabao-Sun @1996fanrui PTAL

File: flink-core/src/test/java/org/apache/flink/types/StringSerializationTest.java
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/types/StringSerializationTest.java:39]:
```suggestion
class StringSerializationTest {
```

File: flink-core/src/test/java/org/apache/flink/types/StringValueSerializationTest.java
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/types/StringValueSerializationTest.java:35]:
```suggestion
class StringValueSerializationTest {
```

File: flink-core/src/test/java/org/apache/flink/types/PrimitiveDataTypeTest.java
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/types/PrimitiveDataTypeTest.java:139]: assertThatComparable(string0).isEqualByComparingTo(string0);

We can use `assertThatComparable` here.

File: flink-core/src/test/java/org/apache/flink/types/PrimitiveDataTypeTest.java
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/types/PrimitiveDataTypeTest.java:146]: assertThatComparable and hasSameHashCode

**Review by 1996fanrui — COMMENTED** [review]
Thanks @GOODBOY008 for the contribution!

Overall is fine for me, I only left some minor comments, please take a look in your free time, thanks~

**Review by 1996fanrui — APPROVED** [review]
Thanks for the quick update!

LGTM

**Comment by 1996fanrui** [comment]
Hi @GOODBOY008 , would you mind rebasing master branch before merging? It can ensure the master branch works well after merging.

Also, I will merge this PR if no other comments within 2 days.

cc @Jiabao-Sun

**Review by Jiabao-Sun — APPROVED** [review]
Thanks @GOODBOY008 for this great work and @1996fanrui for the hard review.
LGTM.

**Pull request merged**
